Add a link to the RFC dashboard in the README

anaelleltd opened the pull request:
The status of each RFC is now tracked [here](https://spiced-longship-f1a.notion.site/abbeb769972f4aa0afbfe41cac4544f1?v=b8d75b58aec249ccb545ce68ffb70922).

acatangiu reviewed (approved):
nice!

how is this list maintained (and updated)?

anaelleltd commented:
> how is this list maintained (and updated)?

The status of each RFC can only be manually updated, because there is currently no way to know if an RFC is WIP or implemented without confirmation from the author(s). I will share a brief about this process in the Fellowship-only channel next week.

I did a manual data import (via csv) within the Notion page for now. I am hoping to link the _Polkadot Fellowship_ repo with Notion so that the PRs can be automatically fetched and updated in the RFC dashboard, as demoed [here](https://www.notion.so/integrations/github). However, I need to get the required OAuth permissions first, and this could be tricky because Notion just wants to access everything from connected GitHub organisations. 😩
